server/src/repositories/stack.repository.ts

```diff
@@ -148,6 +148,7 @@ export class StackRepository {
       assetIds.push(...stack.assets.map(({ id }) => id));
     }
 
+    await this.db.deleteFrom('asset_stack').where('id', 'in', ids).execute();
```

Comment:
Surely this could be after the next line otherwise none of the assets will be updated since it will be filtering on stacks that no longer exist